pkg/stats/template.go

```diff
@@ -0,0 +1,80 @@
+package stats
+
+import (
+	"fmt"
+	"io"
+	"text/template"
+	"time"
+
+	"github.com/fastly/go-fastly/fastly"
+	"github.com/mitchellh/mapstructure"
+)
+
+var blockTemplate = template.Must(template.New("stats_block").Parse(
+	`Service ID:         {{ .ServiceID }}
+Start Time:         {{ .StartTime }}
+--------------------------------------------------
+Hit Rate:           {{ .HitRate }}
+Avg Hit Time:       {{ .AvgHitTime }}
+Avg Miss Time:      {{ .AvgMissTime }}
+
+Request BW:         {{ .RequestBytes }}
+  Headers:          {{ .RequestHeaderBytes }}
+  Body:             {{ .RequestBodyBytes }}
+
+Response BW:        {{ .ResponseBytes }}
+  Headers:          {{ .ResponseHeaderBytes }}
+  Body:             {{ .ResponseBodyBytes }}
+
+Requests:           {{ .RequestCount }}
+  Hit:              {{ .Hits }}
+  Miss:             {{ .Miss }}
+  Pass:             {{ .Pass }}
+  Synth:            {{ .Synth }}
+  Error:            {{ .Errors }}
+  Uncacheable:      {{ .Uncacheable }}
+
+`))
+
+func fmtBlock(out io.Writer, service string, block statsResponseData) error {
+	var agg fastly.Stats
+	if err := mapstructure.Decode(block, &agg); err != nil {
+		return err
+	}
+
+	hitRate := 0.0
+	if agg.Hits > 0 {
+		hitRate = float64((agg.Hits - agg.Miss - agg.Errors)) / float64(agg.Hits)
+	}
+
+	// TODO: parse the JSON more strictly so this doesn't need to be dynamic.
```

Comment:
Noting the TODO here, do you want to do this as part of this PR or at a later date?

Reply:
Let's do this later. I need to think about whether we can unify these APIs at all.